### DISCUSSION TITLE
Only onSuccess gets fired even if error has occurred in useMutation. WHY?

### DISCUSSION THREAD

**Discussion by acharyayush** [TanStack/query#6626] [Q&A]
//mutation hook
const { data, mutate: mutateCustomer } = useMutation({
mutationFn: createCustomer,
onSuccess: () => {
console.log("navigated")
navigate("/")
},
onError: () => {
console.log("error occured")
},
})
//api.js
//i have BASEURL
export const createCustomer = async (customerDetail) => {
try {
const { data } = await axios.post(
${BASEURL}/api/65865d8b75a727705c869123/customers/add,
customerDetail
)
toast("success", "New customer is created 😁")
return data
} catch (err) {
toast("error", err.message)
return err
}
}
if i submit with empty customer details, i get internal server error but onSuccess gets triggered and page is navigated to homepage, onError never gets triggerred...

**Comment by TkDodo** [comment]
This code turns the error into a resolved promise 

```
} catch (err) {
toast("error", err.message)
return err
}
```

https://tkdodo.eu/blog/react-query-fa-qs#logging

**Comment by acharyayush** [comment]
Thanks a lot.😌
